src/features/courses/course-route.ts

```diff
@@ -307,6 +309,78 @@ router.get('/topic/:topicId/version/:userId',
     })
 );
 
+router.put('/topic/:topicId/endExport', 
+    // this call is expected from a microservice, so doesn't go through authentication
+    validate(endBulkExportValidation),
+    // This is due to a typescript issue where the type mismatches extractMap
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    asyncHandler(async (req: RederlyExpressRequest<any, unknown, EndBulkExportRequest.body, EndBulkExportRequest.query>, _res: Response, next: NextFunction) => {
+        const topic = await CourseTopicContent.findOne({
+            where: {
+                id: req.params.topicId,
+                active: true,
+            },
+        });
+
+        if (_.isNil(topic)) {
+            throw new NotFoundError('A endExport PUT was received for a non-existant topic. Was it deleted?');
```

Comment:
An*? :trollface:

Reply:
This error would get shown on the frontend, we might want to clean up the message

Reply:
nvm, it's called from the microservice*